replica/table.cc

```diff
@@ -196,7 +196,10 @@ table::add_memtables_to_reader_list(std::vector<flat_mutation_reader_v2>& reader
     // point queries can be optimized as they span a single compaction group.
     if (range.is_singular() && range.start()->value().has_key()) {
         const dht::ring_position& pos = range.start()->value();
-        auto sg = storage_group_for_token(pos.token());
+        auto* sg = storage_group_for_token(pos.token());
+        if (!sg) {
+            return;
+        }
```

Comment:
Does this fix a bug? Did `storage_group_for_token()` not return `null` before this patch?

Reply:
no, we can just change storage_group_for_token() to return a reference, and throw an exception if storage group is not allocated, which is a bug as read reached this replica after the tablet replica left it. coordinator should have drained reads of old replica before tablet transitions into reading new replica only, but it's important that we handle it at replica side too, rather than blindly ignoring it.

Reply:
`storage_group_for_token` did not return `null` before this patch.
I'm not sure that actually fixed a bug if tablet cleanup post migration is done properly with invalidating the memtable and row-cache, so even if a lingering read tries to read from the tablet replica after it moved away will get to the storage_group, it would be empty by now so it would find nothing.